acp/phpbb_ideas_module.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\acp;
+
+class phpbb_ideas_module
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var string */
+	public $u_action;
+
+	public function main($id, $mode)
+	{
+		global $config, $db, $phpbb_log, $request, $template, $user;
+
+		$this->config = $config;
+		$this->db = $db;
+		$this->log = $phpbb_log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+
+		// Add the board rules ACP lang file
+		$this->user->add_lang_ext('phpbb/ideas', 'phpbb_ideas_acp');
+
+		// Load a template from adm/style for our ACP page
+		$this->tpl_name = 'acp_phpbb_ideas';
+
+		// Set the page title for our ACP page
+		$this->page_title = 'ACP_PHPBB_IDEAS_SETTINGS';
+
+		// Define the name of the form for use as a form key
+		$form_name = 'acp_phpbb_ideas_settings';
+		add_form_key($form_name);
+
+		// Set an empty errors array
+		$errors = array();
+
+		$display_vars = array(
+			'ideas_forum_id'	=> array('lang' => 'ACP_IDEAS_FORUM_ID',	'validate' => 'string',	'type' => 'custom', 'method' => 'select_ideas_forum', 'explain' => true),
+			'ideas_poster_id'	=> array('lang' => 'ACP_IDEAS_POSTER_ID',	'validate' => 'string',	'type' => 'custom', 'method' => 'select_ideas_topics_poster', 'explain' => true),
+		);
+
+		$this->new_config = $config;
+		$cfg_array = ($request->is_set('config')) ? $request->variable('config', array('' => ''), true) : $this->new_config;
+		$submit = (isset($_POST['submit'])) ? true : false;
```

Comment:
use `$submit = $this->request->is_set('submit');` and since it returns a boolean, don't need the conditional true or false part.